config/config_test.go

```diff
@@ -0,0 +1,164 @@
+			prepFunc: func(configPath string) error {
+				config := Config{
+					DefaultSort:       "Name",
+					Columns:           []string{"Name", "Size"},
+					OllamaAPIKey:      "testkey",
+					OllamaAPIURL:      "http://testurl",
+					LMStudioFilePaths: "/test/path",
+					LogLevel:          "debug",
+					LogFilePath:       "/test/path/gollama.log",
+					SortOrder:         "name",
+					StripString:       "strip",
+					Editor:            "nano",
+					DockerContainer:   "testcontainer",
+				}
+				return saveTempConfig(configPath, config)
+			},
+			expected: Config{
+				DefaultSort:       "Name",
+				Columns:           []string{"Name", "Size"},
+				OllamaAPIKey:      "testkey",
+				OllamaAPIURL:      "http://testurl",
+				LMStudioFilePaths: "/test/path",
+				LogLevel:          "debug",
+				LogFilePath:       "/test/path/gollama.log",
+				SortOrder:         "name",
+				StripString:       "strip",
+				Editor:            "nano",
+				DockerContainer:   "testcontainer",
+				LastSortSelection: "name",
+			},
+			expectedError: false,
+		},
+		{
+			name: "Config file does not exist",
+			prepFunc: func(configPath string) error {
+				return nil // No prep needed for this test case
+			},
+			expected:      defaultConfig,
+			expectedError: false,
+		},
+		{
+			name: "Config file is invalid",
+			prepFunc: func(configPath string) error {
+				return os.WriteFile(configPath, []byte("invalid json"), 0644)
+			},
+			expected:      Config{},
+			expectedError: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			configPath := getConfigPath()
+			os.Remove(configPath) // Ensure config file does not exist at start
+			if tt.prepFunc != nil {
+				if err := tt.prepFunc(configPath); err != nil {
+					t.Fatalf("prepFunc failed: %v", err)
+				}
+			}
+
+			got, err := LoadConfig()
+			if (err != nil) != tt.expectedError {
+				t.Errorf("LoadConfig() error = %v, expectedError %v", err, tt.expectedError)
+				return
+			}
+			if !tt.expectedError && !compareConfigs(got, tt.expected) {
+				t.Errorf("LoadConfig() got = %v, want %v", got, tt.expected)
+			}
+		})
+	}
+}
+
+func TestSaveConfig(t *testing.T) {
+	tests := []struct {
+		name          string
+		input         Config
+		expectedError bool
+	}{
+		{
+			name: "Valid config save",
+			input: Config{
+				DefaultSort:       "Name",
+				Columns:           []string{"Name", "Size"},
+				OllamaAPIKey:      "testkey",
+				OllamaAPIURL:      "http://testurl",
+				LMStudioFilePaths: "/test/path",
+				LogLevel:          "debug",
+				LogFilePath:       "/test/path/gollama.log",
+				SortOrder:         "name",
+				StripString:       "strip",
+				Editor:            "nano",
+				DockerContainer:   "testcontainer",
+			},
+			expectedError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			configPath := getConfigPath()
+			os.Remove(configPath) // Ensure config file does not exist at start
```

Comment:
I think instead here, the test should be broken up into something like the following using a _custom temporary path_ for saving and loading the default config when tests are run.

1. When I generate the default config, does it get created?
2. Is it valid json and does it contain the correct defaults?
3. When using the default config AND only if the Ollama process is running (as it won't be in CI), can I connect to Ollama (e.g. list models)?